backend/internal/ou/store/store.go

```diff
@@ -0,0 +1,447 @@
+	if err != nil {
+		logger.Error("Failed to get organization unit groups", log.Error(err))
+		return model.OrganizationUnit{}, fmt.Errorf("failed to get organization unit groups: %w", err)
+	}
+	ou.Groups = *groups
+
+	return ou, nil
+}
+
+// UpdateOrganizationUnit updates an existing organization unit.
+func UpdateOrganizationUnit(ou model.OrganizationUnit) error {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+
+	dbClient, err := provider.NewDBProvider().GetDBClient("identity")
+	if err != nil {
+		logger.Error("Failed to get database client", log.Error(err))
+		return fmt.Errorf("failed to get database client: %w", err)
+	}
+	defer func() {
+		if closeErr := dbClient.Close(); closeErr != nil {
+			logger.Error("Failed to close database client", log.Error(closeErr))
+		}
+	}()
+
+	// Update the organization unit
+	_, err = dbClient.Execute(
+		QueryUpdateOrganizationUnit,
+		ou.ID,
+		ou.Parent,
+		ou.Name,
+		ou.Description,
+	)
+	if err != nil {
+		logger.Error("Failed to execute update organization unit query", log.Error(err))
+		return fmt.Errorf("failed to execute query: %w", err)
+	}
+
+	return nil
+}
+
+// DeleteOrganizationUnit deletes an organization unit.
+func DeleteOrganizationUnit(id string) error {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+
+	dbClient, err := provider.NewDBProvider().GetDBClient("identity")
+	if err != nil {
+		logger.Error("Failed to get database client", log.Error(err))
+		return fmt.Errorf("failed to get database client: %w", err)
+	}
+	defer func() {
+		if closeErr := dbClient.Close(); closeErr != nil {
+			logger.Error("Failed to close database client", log.Error(closeErr))
+		}
+	}()
+
+	_, err = dbClient.Execute(QueryDeleteOrganizationUnit, id)
+	if err != nil {
+		logger.Error("Failed to execute delete organization unit query", log.Error(err))
+		return fmt.Errorf("failed to execute query: %w", err)
+	}
+
+	return nil
+}
+
+// Helper function for executing queries that return string arrays
+func executeQueryForStringArray(
+	query dbmodel.DBQuery, fieldName string, params ...interface{},
+) (*[]string, error) {
+	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, loggerComponentName))
+
+	dbClient, err := provider.NewDBProvider().GetDBClient("identity")
+	if err != nil {
+		logger.Error("Failed to get database client", log.Error(err))
+		return nil, fmt.Errorf("failed to get database client: %w", err)
+	}
+	defer func() {
+		if closeErr := dbClient.Close(); closeErr != nil {
+			logger.Error("Failed to close database client", log.Error(closeErr))
+		}
+	}()
+
+	results, err := dbClient.Query(query, params...)
+	if err != nil {
+		logger.Error("Failed to execute query", log.Error(err))
+		return nil, fmt.Errorf("failed to execute query: %w", err)
+	}
+
+	values := make([]string, 0)
+	for _, row := range results {
+		if val, ok := row[fieldName].(string); ok {
```

Comment:
We're only storing string values right? If so, should we log an error if the string cast failed?

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/7c57d0152029bf8e0495d2c548d693d5be33bd33..0e9b3e18d5ee3f230ea5782b6ed004299e6febe8